src/base.config.ts

```diff
@@ -35,6 +35,7 @@ export abstract class Updater<S extends { id: string } = { id: string }, T exten
 
   abstract assertConfig(config: unknown): asserts config is S;
   abstract prepareNewData(oldData: T | null): T;
+  abstract invalidatePath(): string | undefined;
```

Comment:
if you make this `invalidatePath?(): string` you can leave it unimplemented in the sub classes